File: Guia2-DSM-BF180436/Guia2_Ejercicio1/app/src/main/java/sv/edu/udb/guia2_ejercicio1/MainActivity.java
package sv.edu.udb.guia2_ejercicio1;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    private TextView lblresultado;
    public int contador = 0;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        lblresultado = findViewById(R.id.lblNum);
        if (savedInstanceState != null) {
            contador = savedInstanceState.getInt("contador");
            lblresultado.setText(String.valueOf(contador));
        }
    }

    public void sumar(View view) {
        contador++;
        if (contador > 9) contador = 0;

        lblresultado.setText(String.valueOf(contador));
    }

    @Override
    protected void onSaveInstanceState(Bundle estado) {
        super.onSaveInstanceState(estado);
        estado.putInt("contador", contador);
    }

    @Override
    protected void onRestoreInstanceState(Bundle estado) {
        super.onRestoreInstanceState(estado);
        contador = estado.getInt("contador");
        lblresultado.setText(String.valueOf(contador));
    }
}

File: Desafio1-DSM-BF180436/app/src/main/java/sv/edu/udb/desafio1_bf180436/MainActivity.kt
package sv.edu.udb.desafio1_bf180436

import android.content.res.Configuration
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.os.PersistableBundle
import android.view.MenuItem
import android.widget.Toast
import android.widget.Toolbar
import androidx.appcompat.app.ActionBarDrawerToggle
import androidx.appcompat.app.AppCompatDialogFragment
import androidx.core.view.GravityCompat
import androidx.drawerlayout.widget.DrawerLayout
import androidx.fragment.app.Fragment
import com.google.android.material.navigation.NavigationView

class MainActivity : AppCompatActivity(), NavigationView.OnNavigationItemSelectedListener {

    private lateinit var drawer: DrawerLayout
    private lateinit var toggle: ActionBarDrawerToggle

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        var toolbar: androidx.appcompat.widget.Toolbar = findViewById(R.id.toolbar_main)
        setSupportActionBar(toolbar)

        drawer = findViewById(R.id.drawer_layout)

        toggle = ActionBarDrawerToggle(this,drawer, toolbar, R.string.navigation_drawer_open,R.string.navigation_drawer_close)
        drawer.addDrawerListener(toggle)

        supportActionBar?.setDisplayHomeAsUpEnabled(true)
        supportActionBar?.setHomeButtonEnabled(true)

        val navigationView: NavigationView = findViewById(R.id.nav_view)
        navigationView.setNavigationItemSelectedListener(this)
    }

    override fun onNavigationItemSelected(item: MenuItem): Boolean {
        item.isChecked = true
        when (item.itemId) {
            R.id.nav_ejercicio_1 -> replacefragment(ejercicio1Fragment(), item.title.toString())
            R.id.nav_ejercicio_2 -> replacefragment(ejercicio2Fragment(), item.title.toString())
            R.id.nav_ejercicio_3 -> replacefragment(ejercicio3Fragment(), item.title.toString())
            R.id.nav_logout -> finish()
        }

        drawer.closeDrawer(GravityCompat.START)
        return true
    }

    private fun replacefragment(fragment: Fragment, title: String) {
        val frangmentManager = supportFragmentManager
        val frangmentTransaction = frangmentManager.beginTransaction()
        frangmentTransaction.replace(R.id.frameLayout,fragment)
        frangmentTransaction.commit()
        drawer.closeDrawers()
        setTitle(title)
    }

    override fun onPostCreate(savedInstanceState: Bundle?) {
        super.onPostCreate(savedInstanceState)
        toggle.syncState()
    }

    override fun onConfigurationChanged(newConfig: Configuration) {
        super.onConfigurationChanged(newConfig)
        toggle.onConfigurationChanged(newConfig)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (toggle.onOptionsItemSelected(item)) {
            return true
        }

        return super.onOptionsItemSelected(item)
    }
}